File: packages/zynth-keyboard/android/ZynthKeyboard/src/main/java/dev/zynth/keyboard/ZynthKeyboard.kt
package dev.zynth.keyboard

import android.app.Activity
import com.zynth.kit.runtime.ZynthRuntime

/**
 * Public interface for ZynthKeyboard module
 */
object ZynthKeyboard {
    private var moduleInstance: ZynthKeyboardModule? = null

    /**
     * Initialize the keyboard module with an Activity and ZynthRuntime instance
     * Should be called during app initialization before the JS bundle loads
     */
    @JvmStatic
    fun initialize(activity: Activity, runtime: ZynthRuntime) {
        if (moduleInstance != null) {
            android.util.Log.w("ZynthKeyboard", "Module already initialized")
            return
        }

        moduleInstance = ZynthKeyboardModule(activity, runtime)
        android.util.Log.d("ZynthKeyboard", "Module initialized")
    }

    /**
     * Clean up the module (called during app teardown)
     */
    @JvmStatic
    fun cleanup() {
        moduleInstance?.onDestroy()
        moduleInstance = null
    }
}

File: packages/zynth-keyboard/ios/ZynthKeyboard.swift
//
//  ZynthKeyboard.swift
//  ZynthKeyboard
//
//  Public interface for ZynthKeyboard module
//

import Foundation
import ZynthKit

@objc(ZynthKeyboard)
public class ZynthKeyboard: NSObject {
  
  private static var moduleInstance: ZynthKeyboardModule?
  
  /// Initialize the keyboard module with a ZynthRuntime instance
  /// Should be called during app initialization before the JS bundle loads
  @objc public static func initialize(with runtime: ZynthRuntime) {
    guard moduleInstance == nil else {
      print("[ZynthKeyboard] Module already initialized")
      return
    }
    
    // Use the static initialize method from ZynthKeyboardModule
    moduleInstance = ZynthKeyboardModule.initialize(with: runtime)
    print("[ZynthKeyboard] Module initialized")
  }
  
  /// Clean up the module (called during app teardown)
  @objc public static func cleanup() {
    moduleInstance = nil
  }
}
